tools/toolchains/__init__.py

```diff
@@ -775,9 +778,22 @@ def compile_command(self, source, object, includes):
 
         return None
 
+    @abstractmethod
+    def parse_dependencies(self, dep_path):
+        """Take in a dependency file generated by the compiler and build a list of
+           all files that the dep_path depends on.
+        """
+        pass
```

Comment:
I was considering that, but I also tested it by deleting one of these methods from a toolchain. It throws an exception when you try to instantiate the object despite the lack of NotImplemented. I think that NotImplemented will provide extra safety: It will prevent toolchains from calling these methods in the subclass. I'm going to replace all one line passes with raise NotImplemented in a few minutes.